proofs/alf/cvc5/programs/Strings.smt3

```diff
@@ -630,3 +682,158 @@
+                                          (str.in_re (str.++ s1 s2) (re.++ r1 r2))))) ; otherwise stuck
+          (@re.empty                  ($str_re_consume_rec (str.++ s1 s2) r2 @result.null rev)) ; finished current component
+          ((re.range s3 s5)           (alf.ite (alf.and (string_check_length_one s1) ($str_is_char_range s3 s5))
+                                        (alf.ite ($str_eval_str_in_re s1 (re.range s3 s5))
+                                          ($str_re_consume_rec s2 r2 @result.null rev)
+                                          false)
+                                        (str.in_re (str.++ s1 s2) (re.++ r1 r2))))
+          (re.allchar                 (alf.ite (string_check_length_one s1)
+                                        ($str_re_consume_rec s2 r2 @result.null rev)
+                                        (str.in_re (str.++ s1 s2) (re.++ r1 r2))))
+          ((re.* r3)                  ; see what happens if we unroll once
+                                      (alf.match ((sr String) (rr RegLan))
+                                        ($str_re_consume_rec (str.++ s1 s2) ($re_to_flat_form r3 rev) @result.null rev)
+                                        (
+                                          ; We can't unroll even once, thus we must skip it.
+                                          (false                     ($str_re_consume_rec (str.++ s1 s2) r2 @result.null rev))
+                                          ; If we fully consumed, now we go back and check if we get a conflict if we skip.
+                                          ((str.in_re sr @re.empty)  (alf.match ((br Bool))
+                                                                        ($str_re_consume_rec (str.++ s1 s2) r2 @result.null rev)
+                                                                        (
+                                                                          ; Skipping would give a conflict.
+                                                                          (false (alf.ite (alf.is_eq (str.++ s1 s2) sr)
+                                                                                    ; if we did not consume anything, we are stuck.
+                                                                                    (str.in_re (str.++ s1 s2) (re.++ r1 r2))
+                                                                                    ; otherwise, we continue with the RE consumed once.
+                                                                                    ($str_re_consume_rec sr (re.++ r1 r2) @result.null rev)
+                                                                                  ))
+                                                                          ; Otherwise we are stuck.
+                                                                          (br    (str.in_re (str.++ s1 s2) (re.++ r1 r2)))
```

Comment:
The problem is that this introduces search, since it not clear whether we should consume or not. This method is essentially only following cases that are implied, e.g. if it is impossible to consume one unfolding, or if it impossible to skip. If both are possible, we are stuck since otherwise this algorithm would have exponential runtime.